recipes-flexray-utils/flexraywd/files/flexray_boot.sh

```diff
@@ -3,7 +3,42 @@ WORKING_DIR="/opt/hm"
 
 FIRST_TIME_BOOT_FILE=/usr/bin/first_boot_after_update_flexray.txt
 NO_FLEXRAY_FILE=/etc/no_flexray_upgrade
-PREFIX=mx4-msg-flexray-update
+PREFIX=mx4-msg-flexray
+
+# start and restart flexraytimesync utility will syncronize the time in the MX4 with the time used in the FlexRay slave node for timestamping frames.
+function start_flexraytimesync()
+{
+
+	while true; do
+		logger -t ${PREFIX} "Start flexraytimesync"
+		/usr/bin/flexraytimesync
+		logger -t ${PREFIX} "Stopped flexraytimesync exit:$?"
+		sleep 1
+	done
+}
+# start and restart flexrayd demon for udp package to vflexray0.
+function start_flexrayd()
+{
+
+	while true; do
+		logger -t ${PREFIX} "Start flexrayd"
+		/usr/bin/flexrayd
+		logger -t ${PREFIX} "Stopped flexrayd exit:$?"
+		sleep 1
+	done
+}
+# abort script remove start programs.
+function Stop_script ()
+{
+	kill -0 "$flexraytimesync_PID"
+	if [[ $? == 0 ]]; then
+		kill -9 "$flexraytimesync_PID" &
+	fi
+	kill -0 "$flexrayd_PID"
+	if [[ $? == 0 ]]; then
+		kill -9 "$flexrayd_PID" &
+	fi
+}
 
```

Comment:
How about checking for the existence of a run file somewhere in /var/ and stop processes by deleting it? Killing the processes could be done as a last resort thing.

Reply:
It would require to change those applications so they can only run on a single instans, right now they can be used multiple times. I think your idea is good but is it needed in this case. the applications shall run untill restart of unit or suspend?

Reply:
OK, I see. Then I suppose this way is acceptable.